blackbox/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/adamplansky/go-bridge-mentoring/blackbox/blackbox"
+)
+
+func run() error {
+	jobs := blackbox.ParseInput()
+	logReporter, err := blackbox.NewLogResult()
+	if err != nil {
+		return err
+	}
+	s, err := blackbox.NewScraper(jobs, logReporter)
+	if err != nil {
+		return err
+	}
+
+	resultCh := s.RunReporter()
+	defer close(resultCh)
+	s.Scrape(resultCh)
+
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
+	<-sig
+
```

Comment:
you may catch panic on while closing app - "publish to closed channel" as you are not actually stopping scrapers